httpd/http_server.c

```diff
@@ -0,0 +1,333 @@
+        // files to the client
+        // if this is not an API call we will abort this handler later
+        // and let the internal serve file handler of h2o care for things
+
+        if (end_loc == SIZE_MAX) {
+            req->path.len = 1;
+            req->path_normalized.len = 1;
+        } else {
+            size_t offset = norm_path.base - req->path_normalized.base;
+            req->path.len -= offset;
+            req->path.base += offset;
+            req->query_at -= offset;
+            req->path_normalized.len -= offset;
+            req->path_normalized.base += offset;
+        }
+    }
+
+    // workaround for a dashboard bug which causes sometimes urls like
+    // "//api/v1/info" to be caled instead of "/api/v1/info"
+    if (norm_path.len > 2 &&
+        norm_path.base[0] == '/' &&
+        norm_path.base[1] == '/' ) {
+            norm_path.base++;
+            norm_path.len--;
+    }
+
+    size_t api_loc = h2o_strstr(norm_path.base, norm_path.len, H2O_STRLIT(API_V1_PREFIX));
+    if (api_loc == SIZE_MAX)
+        return 1;
+
+    h2o_iovec_t api_command = norm_path;
+    api_command.base += api_loc + strlen(API_V1_PREFIX);
+    api_command.len -= api_loc + strlen(API_V1_PREFIX);
+
+    if (!api_command.len)
+        return 1;
+
+    // this (emulating struct web_client) is a hack and will be removed
+    // in future PRs but needs bigger changes in old http_api_v1
+    // we need to make the web_client_api_request_v1 to be web server
+    // agnostic and remove the old webservers dependency creep into the
+    // individual response generators and thus remove the need to "emulate"
+    // the old webserver calling this function here and in ACLK
+    struct web_client w;
+    w.response.data = buffer_create(NBUF_INITIAL_SIZE_RESP, NULL);
+    w.response.header = buffer_create(NBUF_INITIAL_SIZE_RESP, NULL);
+    w.decoded_query_string[0] = 0;
+    w.acl = WEB_CLIENT_ACL_DASHBOARD;
+
+    char *path_c_str = iovec_to_cstr(&api_command);
+    char *path_unescaped = url_unescape(path_c_str);
+    freez(path_c_str);
+
+    IF_HAS_URL_PARAMS(req) {
+        h2o_iovec_t query_params = URL_PARAMS_IOVEC_INIT_WITH_QUESTIONMARK(req);
+        char *query_c_str = iovec_to_cstr(&query_params);
+        char *query_unescaped = url_unescape(query_c_str);
+        freez(query_c_str);
+        strcpy(w.decoded_query_string, query_unescaped);
+        freez(query_unescaped);
+    }
+
+    web_client_api_request_v1(*host, &w, path_unescaped);
+    freez(path_unescaped);
+
+    h2o_iovec_t body = buffer_to_h2o_iovec(w.response.data);
+
+    // we move msg body to req->pool managed memory as it has to
+    // live until whole response has been encrypted and sent
+    // when req is finished memory will be freed with the pool
+    void *managed = h2o_mem_alloc_shared(&req->pool, body.len, NULL);
+    memcpy(managed, body.base, body.len);
+    body.base = managed;
+
+    req->res.status = 200;
```

Comment:
Now about this `200` and previous `400`, please, let us use our internal [aliases](https://github.com/netdata/netdata/blob/513452aa447759acecb16504673db7dce40ffaa2/web/server/web_client.h#L12-L33) for them, or to use H2O aliases if they have.